core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -594,13 +594,21 @@ class KafkaServer(
               val clientResponse = NetworkClientUtils.sendAndReceive(networkClient, request, time)
 
               val shutdownResponse = clientResponse.responseBody.asInstanceOf[ControlledShutdownResponse]
-              if (shutdownResponse.error == Errors.NONE && shutdownResponse.data.remainingPartitions.isEmpty) {
+              if (shutdownResponse.error != Errors.NONE) {
+                info(s"Controlled shutdown request returned after ${clientResponse.requestLatencyMs}ms " +
+                  s"with error ${shutdownResponse.error}")
```

Comment:
Should we log as error level or warn level?

Reply:
Hmm, I think INFO is ok. Retries in some cases are not uncommon. It's only an error if we unclean shutdown after all retries have failed.